postprocess/postprocess.go

```diff
@@ -0,0 +1,161 @@
+// package postproces contains functionality to postprocess pointcloud maps
+package postprocess
+
+import (
+	"bytes"
+	"errors"
+	"image/color"
+	"reflect"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/pointcloud"
+)
+
+type Instruction int
+
+const (
+	Add    Instruction = iota
+	Remove             = iota
+)
+
+const (
+	// expected strings for DoCommand
+	ToggleCommand = "postprocess_toggle"
+	AddCommand    = "postprocess_add"
+	RemoveCommand = "postprocess_remove"
+	UndoCommand   = "postprocess_undo"
+)
+
+var (
+	// ErrBadPostprocessingPointsFormat denotes that the points have not been properly formatted
+	ErrBadPostprocessingPointsFormat = errors.New("could not parse provided points")
+)
+
+type Task struct {
+	Instruction Instruction
+	Points      []r3.Vector
+}
+
+func ParseDoCommand(
+	unstructuredPoints interface{},
+	instruction Instruction,
+) (Task, error) {
+	if reflect.TypeOf(unstructuredPoints).Kind() != reflect.Slice {
+		return Task{}, ErrBadPostprocessingPointsFormat
+	}
+
+	task := Task{}
+	slice := reflect.ValueOf(unstructuredPoints)
+
+	for i := 0; i < slice.Len(); i++ {
+		val := slice.Index(i).Elem()
+		if reflect.TypeOf(val).Kind() != reflect.Struct {
+			return Task{}, ErrBadPostprocessingPointsFormat
+		}
+
+		x := val.MapIndex(reflect.ValueOf("X"))
+		y := val.MapIndex(reflect.ValueOf("Y"))
+		task.Points = append(task.Points, r3.Vector{X: x.Elem().Float(), Y: y.Elem().Float()})
+	}
+
+	task.Instruction = instruction
+	return task, nil
+}
```

Comment:
This is all somewhat annoying/janky because the `DoCommand` expects an `interface{}` and I want a somewhat complex data structure as input (slice of maps) and it will panic if the input doesn't conform. Any other/better ideas on how to do this would be great